File: pkg/java/src/main/java/dev/openfga/language/FgaModFile.java
package dev.openfga.language;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;

public class FgaModFile {

    public static final String JSON_PROPERTY_SCHEMA = "schema";
    private ModFileStringProperty schema;

    public static final String JSON_PROPERTY_CONTENTS = "contents";
    private ModFileArrayProperty contents;

    public FgaModFile() {}

    public FgaModFile schema(ModFileStringProperty schema) {
        this.schema = schema;
        return this;
    }

    @JsonProperty(JSON_PROPERTY_SCHEMA)
    @JsonInclude(value = JsonInclude.Include.ALWAYS)
    public ModFileStringProperty getSchema() {
        return this.schema;
    }

    public FgaModFile contents(ModFileArrayProperty contents) {
        this.contents = contents;
        return this;
    }

    @JsonProperty(JSON_PROPERTY_CONTENTS)
    @JsonInclude(value = JsonInclude.Include.ALWAYS)
    public ModFileArrayProperty getContents() {
        return this.contents;
    }
}

File: pkg/csharp/src/OpenFga.Language/ModFile/FgaModFile.cs
using System.Text.Json.Serialization;

namespace OpenFga.Language.ModFile;

public class FgaModFile
{
    public const string JsonPropertySchema = "schema";
    public const string JsonPropertyContents = "contents";

    [JsonPropertyName(JsonPropertySchema)]
    public ModFileStringProperty? Schema { get; set; }

    [JsonPropertyName(JsonPropertyContents)]
    public ModFileArrayProperty? Contents { get; set; }

    public FgaModFile() { }

    public FgaModFile SetSchema(ModFileStringProperty? schema)
    {
        Schema = schema;
        return this;
    }

    public FgaModFile SetContents(ModFileArrayProperty? contents)
    {
        Contents = contents;
        return this;
    }
}
